docs/writing_an_aggregate_builder.md

```diff
@@ -122,3 +122,27 @@ class ListAllFilesBuilder implements Builder {
   }
 }
 ```
+
+## Using a `Resolver`
+
+The `Resolver` provided by the build system only works when the primary input to
+a build step is a `.dart` library, and then only for the code transitively
+imported by that library. If an aggregate builder needs to resolve Dart code
+form the inputs it globs then it needs to be split into two steps. Each of these
+steps can be implemented as a separate `builder_factory` (and `Builder` class)
+but roll into the same builder definition in `build.yaml` if they can both
+output to cache - otherwise for the final output to source they need to be
+separate builder definitions in `build.yaml` and use `runs_before` or
```

Comment:
We need to provide an example here I think, otherwise this introduces a lot of things the reader probably hasn't heard of. We could also link to docs for these options to help.

Reply:
I dropped the bit about `runs_before` and added an example for `required_inputs`. I think it's a bit more clear that way.